### PR TITLE
 Unit Test suites for ViewModels

### PR DESCRIPTION
**Pull request by sanyud24**
Tested various MutableLiveData ViewModel classes for the following packages:
- ### package org.opendatakit.services.sync.actions.viewModels;
   - AbsSyncViewModel
   - LoginViewModel
   - SyncViewModel
   - VerifyViewModel
- ### package org.opendatakit.services.preferences;
   - PreferenceViewModel
   - Added **android.arch.core.executor.testing.InstantTaskExecutorRule**:  swaps the background executor used by the Architecture Components with a different one which executes each task synchronously.
   
Screenshot from 2022-06-27 15-37-32

### CODE REVIEW

**Comment by sanyud24** [comment]
Hi, @wbrunette  @Lamouresparus I've finished testing the ViewModel classes.

**Comment by wbrunette** [comment]
@sanyud24 can you fix the merge conflict

File: services_app/src/test/java/org/opendatakit/services/sync/actions/viewModels/viewModels/AbsSyncViewModelTest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    @Test
+    public void checkIfIsFirstLaunched_isAllowed() throws Exception {
+        absSyncViewModel.setIsFirstLaunch(true);
+        assertEquals(true, absSyncViewModel.checkIsFirstLaunch().getValue());
+    }
+
+    @Test
+    public void checkIfServerUrl_isAvailable() throws Exception {
+        String serverUrl = "https://odk-x.org/";
```
Comment (wbrunette) [inline services_app/src/test/java/org/opendatakit/services/sync/actions/viewModels/viewModels/AbsSyncViewModelTest.java:52]: please use constants

File: services_app/src/test/java/org/opendatakit/services/sync/actions/viewModels/viewModels/AbsSyncViewModelTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    @Test
+    public void checkIfUser_isVerified() {
+        absSyncViewModel.setIsUserVerified(true);
+        assertEquals(true, absSyncViewModel.checkIsUserVerified().getValue());
+    }
+
+    @Test
+    public void checkIfLastSyncTime_isVisible() {
+        long time = LocalTime.parse("12:30").getHour();
```
Comment (wbrunette) [inline services_app/src/test/java/org/opendatakit/services/sync/actions/viewModels/viewModels/AbsSyncViewModelTest.java:99]: use timestamp utilities from android library

File: services_app/src/test/java/org/opendatakit/services/sync/actions/viewModels/viewModels/VerifyViewModelTest.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Test
+    public void checkIfVerifyActions_isAvailable() {
+        verifyViewModel.updateVerifyAction(VerifyServerSettingsActions.VERIFY);
+        assertEquals(VerifyServerSettingsActions.VERIFY, verifyViewModel.getCurrentAction());
+    }
+
+    @Test
+    public void checkVerifyType_isVisible() {
+        String verifyType = "verified";
```
Comment (wbrunette) [inline services_app/src/test/java/org/opendatakit/services/sync/actions/viewModels/viewModels/VerifyViewModelTest.java:34]: please use constants

File: services_app/src/test/java/org/opendatakit/services/sync/actions/viewModels/viewModels/LoginViewModelTest.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    @Test
+    public void checkIfLoginActions_isAvailable() {
+        loginViewModel.updateLoginAction(LoginActions.MONITOR_VERIFYING);
+        assertEquals(LoginActions.MONITOR_VERIFYING, loginViewModel.getCurrentAction());
+    }
+
+    @Test
+    public void checkIfFunctionTypes_isAvailable() {
+        loginViewModel.updateFunctionType("function");
```
Comment (wbrunette) [inline services_app/src/test/java/org/opendatakit/services/sync/actions/viewModels/viewModels/LoginViewModelTest.java:34]: please use constants

**Review by wbrunette — CHANGES_REQUESTED** [review]

**Comment by sanyud24** [comment]
Hi @wbrunette,  Should I also create constants for the boolean fields?

**Comment by wbrunette** [comment]
@sanyud24 that is a great question and the answer is it depends. If you are assigning meaning in setting up your tests then yes, but you should name the constant the meaning. If it's simply just true/false then no. We can discuss more at our next meeting because there is more to explain and it's a good to discuss.

**Review by wbrunette — APPROVED** [review]

**Pull request merged**